### PR TITLE
feat(processors): adding exported functions to processors

### PR DESCRIPTION
**Pull request by enudler**

### CODE REVIEW

**Comment by enudler** [comment]
closes #252

File: src/processors/models/processorsManager.js [hunk 1]
```diff
@@ -28,7 +27,10 @@ module.exports.createProcessor = async function (processor) {
 };
 
 module.exports.getAllProcessors = async function (from, limit) {
-    return databaseConnector.getAllProcessors(from, limit);
+    let allProcessors = await databaseConnector.getAllProcessors(from, limit);
+    allProcessors.forEach(processor => {
```
Comment (NivLipetz) [inline src/processors/models/processorsManager.js]: empty loop

File: src/processors/models/processorsManager.js [hunk 2]
```diff
@@ -79,6 +83,31 @@ function generateProcessorNameAlreadyExistsError() {
     return error;
 }
 
+function generateUnprocessableEntityError(message) {
+    const error = new Error(message);
+    error.statusCode = 422;
+    return error;
+}
+function verifyJSAndGetExportedFunctions(src) {
```
Comment (NivLipetz) [inline src/processors/models/processorsManager.js]: add an enter in between the function declarations

File: src/processors/models/processorsManager.js [hunk 3]
```diff
@@ -62,8 +64,10 @@ module.exports.updateProcessor = async function (processorId, processor) {
     }
 
     processor.created_at = oldProcessor.created_at;
-    fileManager.validateJavascriptContent(processor.javascript);
+    let exportedFunctions = verifyJSAndGetExportedFunctions(processor.javascript);
+    processor.exported_functions = exportedFunctions;
     await databaseConnector.updateProcessor(processorId, processor);
+    processor.exported_functions = verifyJSAndGetExportedFunctions(processor.javascript, true);
```
Comment (NivLipetz) [inline src/processors/models/processorsManager.js]: This isn't needed again. And `verifyJSAndGetExportedFunctions` is accepting only 1 argument in the signature and here you are passing 2

File: src/processors/models/processorsManager.js [hunk 4]
```diff
@@ -79,6 +83,31 @@ function generateProcessorNameAlreadyExistsError() {
...
+        m._compile(src, 'none');
+        let exports = m.exports;
+        exportedFunctions = Object.keys(exports);
+    } catch (err) {
+        let error = generateUnprocessableEntityError('javascript syntax validation failed with error: ' + err.message);
+        throw error;
+    }
+
+    if (exportedFunctions.length === 0) {
+        let error = generateUnprocessableEntityError('javascript has 0 exported function');
```
Comment (NivLipetz) [inline src/processors/models/processorsManager.js]:
```suggestion
        let error = generateUnprocessableEntityError('javascript has 0 exported functions');
```

File: src/processors/models/processorsManager.js [hunk 5]
```diff
@@ -79,6 +83,31 @@ function generateProcessorNameAlreadyExistsError() {
     return error;
 }
 
+function generateUnprocessableEntityError(message) {
```
Comment (NivLipetz) [inline src/processors/models/processorsManager.js]: we have 4 `generateErrorFunctions` in this file. Refactor to `./errors.js` file in the same directory to organize it a bit more

File: src/processors/models/processorsManager.js [hunk 6]
```diff
@@ -28,59 +27,71 @@ module.exports.createProcessor = async function (processor) {
...
-    const error = new Error(ERROR_MESSAGES.NOT_FOUND);
-    error.statusCode = 404;
-    return error;
-}
+function verifyJSAndGetExportedFunctions(src) {
+    let exportedFunctions;
+    try {
+        let m = new module.constructor();
+        m.paths = module.paths;
+        m._compile(src, 'none');
```
Comment (NivLipetz) [inline src/processors/models/processorsManager.js]: from `_compile` documentation (https://github.com/nodejs/node-v0.x-archive/blob/master/lib/module.js#L380) the `'none'` provided to this function is the filename which seems irrelevant so erase that parameter
